src/org/omegat/core/machinetranslators/MTHUBTranslate.java

```diff
@@ -0,0 +1,313 @@
+            this.availableLanguageCodes = getJsonCodes(codesMTHUB);
+
+        } catch (Exception e) {
+                System.err.println("IOException: " + e);
+        }
+    }
+
+    private ArrayList<String> getAvailableLanguageCodes() {
+        return availableLanguageCodes;
+    }
+
+    @Override
+    protected String getPreferenceName() {
+       return Preferences.ALLOW_MTHUB_TRANSLATE;
+    }
+
+    @Override
+    public String getName() {
+        return OStrings.getString("MT_ENGINE_MTHUB");
+    }
+
+    @Override
+    public boolean isConfigurable() {
+        return true;
+    }
+
+    @Override
+    public void showConfigurationUI(Window parent) {
+        MTConfigDialog dialog = new MTConfigDialog(parent, getName()) {
+            @Override
+            protected void onConfirm() {
+                String key = panel.valueField1.getText().trim();
+                boolean temporary = panel.temporaryCheckBox.isSelected();
+                setCredential(PROPERTY_API_KEY, key, temporary);
+            }
+        };
+        dialog.panel.valueLabel1.setText(OStrings.getString("MT_ENGINE_MTHUB_API_KEY_LABEL"));
+        dialog.panel.valueField1.setText(getCredential(PROPERTY_API_KEY));
+        dialog.panel.valueLabel2.setVisible(false);
+        dialog.panel.valueField2.setVisible(false);
+        dialog.panel.temporaryCheckBox.setSelected(isCredentialStoredTemporarily(PROPERTY_API_KEY));
+        dialog.show();
+    }
+
+    @Override
+    protected String translate(Language sLang, Language tLang, String text) throws Exception {
+        if (this.availableLanguageCodes.isEmpty()) {
+         this.setAvailableLanguageCodes();
+        }
+
+        String mthubKey = getCredential(PROPERTY_API_KEY);
+        if (mthubKey == null || mthubKey.isEmpty()) {
+            return OStrings.getString("MT_ENGINE_MTHUB_KEY_NOTFOUND");
+        }
+
+        String prev = getFromCache(sLang, tLang, text);
+        if (prev != null) {
+            return prev;
+        }
+        String charset = "UTF-8";
+
+        try {
+            URLConnection connection = new URL(GT_URL).openConnection();
+            connection.setDoOutput(true);
+            connection.setRequestProperty("Accept-Charset", charset);
+            connection.setRequestProperty("Content-Type", "application/json ;charset=" + charset);
+
+            OutputStream output = connection.getOutputStream();
+
+            ObjectMapper om = new ObjectMapper();
+
+            ArrayList<String> segments = new ArrayList<String>();
+            segments.add(text.substring(0, Math.min(text.length(), LIMIT_CHARACTER - 1)));
+
+            MTHUBTranslateRequest mttr = new MTHUBTranslateRequest(
+                    mthubKey, getNormaliseCode(sLang), getNormaliseCode(tLang), segments);
+
+            String json = om.writeValueAsString(mttr);
+            output.write(json.getBytes(charset));
+
+            Map<String, Object> map = om.readValue(connection.getInputStream(),
```

Comment:
Also the `InputStream` should be used with try-with-resources.

Reply:
One more thing on the `ObjectMapper`: If you're only going to use it to serialize the outgoing payload, you might as well just use Nashorn, which is built-in. See e.g. [`JsonParser`](https://github.com/omegat-org/omegat/blob/master/src/org/omegat/util/JsonParser.java) for how we do this for parsing (we can simply add a `stringify` method).

Reply:
I changed the code. 
I removed List<String> segments, use a TreeMap<String, String> and put the parameters and with WikiGet.post obtain the translation.

Reply:
I don't see any new commits. Did you forget to push, or are you still working on it?

Reply:
I'm still working on it

Reply:
I can't use WikiGet.post method, because when I receive an error like "HTTP/1.1 400 Bad Request", it is different to "HttpURLConnection.HTTP_OK" and I don't receive the json with the error to show to the user. 
I mean, I receive an error from the server and I need have the json to see the error type, but with getStringContent from wikiGet doesn't returne to me because it checked the result is != HttpURLConnection.HTTP_OK
I'm working and when I have a stable version I'll do the push

Reply:
If `WikiGet` doesn't meet your needs then there's no reason to use it. In fact I find it to be a fairly poorly written class to begin with, so I would prefer not to add another piece of code dependent on it, but either way it's up to you.

Reply:
I don't use WikiGet, I created the MTHUBTranslateResponse class to get the response.
MTHUBTranslateResponse response = om.readValue(
                                                  connection.getInputStream(), 
                                                  MTHUBTranslateResponse.class);

I'm going to upload today all the code